src/mudlet-lua/lua/GUIUtils.lua

```diff
@@ -1739,30 +1739,50 @@ function moveCursorDown(window, lines, keep_horizontal)
   if keep_horizontal then x = getColumnNumber(window) end
   moveCursor(window, x, math.min(curLine + lines, getLastLineNumber(window)))
 end
-  
-  -- version of replace function that allows for color, by way of cinsertText
-function creplace(window, text)
-	if not text then text, window = window, nil end
-	window = window or "main"
-	local str, start, stop = getSelection(window)
+
+-- internal function that handles coloured replace variants
+function xReplace(window, text, type)
+  if not text then
+    text = window
+    window = "main"
+  end
+  local str, start, stop = getSelection(window)
 	if window ~= "main" then
 		replace(window, "")
+    moveCursor(window, start, getLineNumber(window))
 	else
 		replace("")
+    moveCursor(start, getLineNumber())
 	end
-	moveCursor(window, start, getLineNumber(window))
-	cinsertText(window, text)
+  if type == 'c' then
+    cinsertText(window, text)
+  elseif type == 'd' then
+    dinsertText(window, text)
+  elseif type == 'h' then
+    echo "test"
```

Comment:
Intended?

Reply:
no, removed and pushed without it. I did test it myself but will get additional testing once I have links